lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1018,16 +1018,36 @@ ClangImporter::create(ASTContext &ctx,
 
   // Set up the file manager.
   {
-    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS;
-    if (!ctx.SearchPathOpts.VFSOverlayFiles.empty() ||
-        importerOpts.ForceUseSwiftVirtualFileSystem) {
-      // If the clang instance has overlays it means the user has provided
-      // -ivfsoverlay options.  We're going to clobber their file system with
-      // the Swift file system, so warn about it.
-      if (!instance.getHeaderSearchOpts().VFSOverlayFiles.empty()) {
-        ctx.Diags.diagnose(SourceLoc(), diag::clang_vfs_overlay_is_ignored);
-      }
+    llvm::IntrusiveRefCntPtr<llvm::vfs::OverlayFileSystem> VFS;
+    if (instance.getHeaderSearchOpts().VFSOverlayFiles.empty()) {
       VFS = ctx.SourceMgr.getFileSystem();
+    } else {
+      // Initialize the clang VFS from its compiler invocation.
+      instance.createFileManager();
+
+      // Create a new overlay file system for the clang importer with the clang
```

Comment:
This logic has the effect of adding the real filesystem to the base of `ClangImporterFS`, even if the Swift filesystem does not have the real filesystem in it. I can't think of any realistic situations where this will break something, but it would be nice to keep the `ClangImporterFS` as close to the Swift filesystem as possible.

My suggestion with `createVFSFromCompilerInvocation` would avoid adding the real filesystem.